sharding/notary/notary.go

```diff
@@ -151,7 +205,287 @@ func joinNotaryPool(client mainchain.Client) error {
 	if err != nil {
 		return fmt.Errorf("unable to deposit eth and become a notary: %v", err)
 	}
+	receipt, err := client.TransactionReceipt(tx.Hash())
+	if err != nil {
+		return err
+	}
+	if receipt.Status == uint(0) {
+
+		return errors.New("Transaction was not successful, unable to deposit ETH and become a notary")
+
+	}
+
+	if inPool, err := isAccountInNotaryPool(client); !inPool || err != nil {
+		if !inPool {
+
+			return errors.New("Account has not been able to be deposited in Notary Pool")
+		}
+
+		return err
+	}
+
 	log.Info(fmt.Sprintf("Deposited %dETH into contract with transaction hash: %s", new(big.Int).Div(sharding.NotaryDeposit, big.NewInt(params.Ether)), tx.Hash().String()))
 
 	return nil
 }
+
+// leaveNotaryPool checks if the account is a Notary in  the SMC.
+// if it is then it checks if it has deposited , if it has it deregisters
+// itself from the pool and the lockup period counts down.
+func leaveNotaryPool(client mainchain.Client) error {
+
+	nreg, err := getNotaryRegistry(client)
+	if err != nil {
+		return err
+	}
+
+	if !nreg.Deposited {
+		return errors.New("Account has not deposited in the Notary Pool")
+	}
+
+	txOps, err := client.CreateTXOpts(nil)
+	if err != nil {
+		return fmt.Errorf("unable to create txOpts: %v", err)
+	}
+	tx, err := client.SMCTransactor().DeregisterNotary(txOps)
+
+	if err != nil {
+		return fmt.Errorf("Unable to deregister Notary: %v", err)
+	}
+
+	receipt, err := client.TransactionReceipt(tx.Hash())
+	if err != nil {
+		return err
+	}
+
+	if receipt.Status == uint(0) {
+
+		return errors.New("Transaction was not successful, unable to deregister Notary")
+
+	}
+
+	if dreg, err := hasAccountBeenDeregistered(client); dreg || err != nil {
+		if !dreg {
+
+			return errors.New("Notary unable to be Deregistered Successfully from Pool")
+		}
+
+		return err
+	}
+
+	log.Info(fmt.Sprintf("Notary Deregistered from the pool with hash: %s", tx.Hash().String()))
+	return nil
+
+}
+
+// releaseNotary releases the Notary from the pool by deleting the notary from
+// the registry and transferring back the despoit
+func releaseNotary(client mainchain.Client) error {
+	nreg, err := getNotaryRegistry(client)
+	if err != nil {
+		return err
+	}
+	if !nreg.Deposited {
+		return errors.New("Account has not deposited in the Notary Pool")
+	}
+
+	if nreg.DeregisteredPeriod == big.NewInt(0) {
+
+		return errors.New("Notary is still registered to the pool")
+
+	}
+
+	if lockup, err := isLockUpOver(client); !lockup || err != nil {
+		if !lockup {
+			return errors.New("Lockup period is not over")
+		}
+		return err
+	}
+
+	txOps, err := client.CreateTXOpts(nil)
+	if err != nil {
+		return fmt.Errorf("unable to create txOpts: %v", err)
+	}
+	tx, err := client.SMCTransactor().ReleaseNotary(txOps)
+
+	if err != nil {
+		return fmt.Errorf("Unable to Release Notary: %v", err)
+	}
+
+	receipt, err := client.TransactionReceipt(tx.Hash())
```

Comment:
So you broadcast the transaction, then immediately check if it has been mined? 

I don't think this will ever work.